packages/notifications/src/redux/reducers/notifications.ts

```diff
@@ -0,0 +1,54 @@
+import { PortalNotificationConfig } from '../../Portal';
+import {
+    ADD_NOTIFICATION,
+    REMOVE_NOTIFICATION,
+    CLEAR_NOTIFICATIONS
+} from '../actions/action-types';
+
+function generateID(type: string): string {
+    return 'cloud-services' + type + '-' + new Date().getTime() + Math.random().toString(36).slice(2);
+}
+
+export interface AddNotificationPayload extends Omit<PortalNotificationConfig, 'id'> {
+    id?: string | number;
+}
+
+const addNotification = (notifications: PortalNotificationConfig[], { payload }: { payload: AddNotificationPayload }): PortalNotificationConfig[] =>
```

Comment:
I would write a new type for the function, because it's a bit complicated to read.